vitals-monitor.mjs

```diff
@@ -1,31 +1,40 @@
+function isVitalInRange(value, min, max) {
+  return value >= min && value <= max;
+}
 
-export async function vitalsOk(temperature, pulseRate, spo2) {
-  if (temperature > 102 || temperature < 95) {
-    console.log("Temperature is critical!");
-    for (let i = 0; i < 6; i++) {
-      process.stdout.write("\r* ");
-      await new Promise(resolve => setTimeout(resolve, 1000));
-      process.stdout.write("\r *");
-      await new Promise(resolve => setTimeout(resolve, 1000));
-    }
-    return false;
-  } else if (pulseRate < 60 || pulseRate > 100) {
-    console.log("Pulse Rate is out of range!");
-    for (let i = 0; i < 6; i++) {
-      process.stdout.write("\r* ");
-      await new Promise(resolve => setTimeout(resolve, 1000));
-      process.stdout.write("\r *");
-      await new Promise(resolve => setTimeout(resolve, 1000));
-    }
-    return false;
-  } else if (spo2 < 90) {
-    console.log("Oxygen Saturation out of range!");
-    for (let i = 0; i < 6; i++) {
-      process.stdout.write("\r* ");
-      await new Promise(resolve => setTimeout(resolve, 1000));
-      process.stdout.write("\r *");
-      await new Promise(resolve => setTimeout(resolve, 1000));
-    }
+function getVitalStatus(temperature, pulseRate, spo2) {
+  const vitalChecks = [
+    { value: temperature, min: 95, max: 102, message: "Temperature is critical!" },
+    { value: pulseRate, min: 60, max: 100, message: "Pulse Rate is out of range!" },
+    { value: spo2, min: 90, max: Infinity, message: "Oxygen Saturation out of range!" },
+  ];
+
+  return vitalChecks

```

Comment:
Nice approach 👍  This statement doesn't need to change even when we add more vitals to check.
Hence it is "closed for modification" and does not need to be re-tested